app/integration/ae-integration/src/main/java/uk/ac/ebi/fg/annotare2/integration/AeIntegrationWatchdog.java

```diff
@@ -380,7 +379,8 @@ public void processSubmitted(Submission submission) throws SubsTrackingException
         }
     }
 
-    private void addFilesToSubstracking(Submission submission, Integer substrackingId, File exportDir) throws SubsTrackingException {
+    @Transactional(rollbackOn = {SubsTrackingException.class})
+    public void addFilesToSubstracking(Submission submission, Integer substrackingId, File exportDir) throws SubsTrackingException {
```

Comment:
Along with this method should we add transactional to `copyDataFiles(submission, substrackingId)` method as well? Because that is the main method which takes a long time to copy files.

Reply:
The `copyDataFiles(submission, substrackingId)` method doesn't have any DB transactions. It gets all the file details from in memory object `Submission`. And as you said it takes long time to process the files so I think it is better to process this outside of transaction so that connections don't have to wait.

Reply:
Okay cool makes sense. Approving now!